xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEStream.cpp

```diff
@@ -671,17 +710,34 @@ bool CActiveAEStreamBuffers::IsDrained()
 
 void CActiveAEStreamBuffers::SetRR(double rr)
 {
-  m_resampleBuffers->SetRR(rr);
+  if (rr == 1.0)
+  {
+    m_resampleBuffers->SetRR(1.0);
+    m_atempoBuffers->SetTempo(1.0);
+  }
+  if (rr < 1.02 && rr > 0.98)
+  {
+    m_resampleBuffers->SetRR(rr);
+    m_atempoBuffers->SetTempo(1.0);
+  }
+  else
+  {
+    m_resampleBuffers->SetRR(1.0);
+    m_atempoBuffers->SetTempo(1.0/rr);
+  }
 }
 
 double CActiveAEStreamBuffers::GetRR()
 {
-  return m_resampleBuffers->GetRR();
+  double tempo = m_resampleBuffers->GetRR();
+  tempo *= m_atempoBuffers->GetTempo();
```

Comment:
Should this be a ``/=``? Looks like when it is set we use the inverse of ``rr``.